src/core/stdcpp/vector.d

```diff
@@ -32,7 +32,7 @@ extern(C++, "std"):
 
 extern(C++, class) struct vector(T, Alloc = allocator!T)
 {
-    import core.lifetime : forward, move, moveEmplace, core_emplace = emplace;
+    import core.lifetime : forward, move, core_emplace = emplace;
```

Comment:
I think it's likely that all of these will trigger the same problem... no?

Reply:
It works fine because UFCS isn't used for those inside vector and outside vector the imports aren't visible.

Reply:
How is it not visible? Those imports are public, and definitely visible.

```D
import core.lifetime : move;

struct Test
{
    import core.lifetime : move;
}

int main()
{
    Test t;
    auto u = move(t); <- works
    auto v = t.move(); <- fails, because sees the local one, and thinks it has no arguments

    return 0;
}
```

Basically, when you import anything into struct/class scope, it seems to be preferred over UFCS.

Reply:
Oh I see, you're saying the imports are private by default. My test is within a module.
Yeah, fair enough, but the whole thing kinda smells to me :/
The same problems may occur within a module, but less likely, and at least you can control the situation in that case.